src/common/guac_rect.h

```diff
@@ -81,5 +103,40 @@ void guac_common_rect_extend(guac_common_rect* rect, const guac_common_rect* min
  */
 void guac_common_rect_constrain(guac_common_rect* rect, const guac_common_rect* max);
 
+/**
+ * Check whether a rectangle intersects another
+ *
+ * @param rect
+ *     Rectangle to check for intersection
+ *
+ * @param master
+ *     The master rectangle
+ *
+ * @return
+ *     -1 if no intersection, 1 if first rect is completely inside the
+ *     master, 0 if partial intersection
+ */
+int guac_common_rect_intersects(guac_common_rect* rect,
+                                const guac_common_rect* master);
+
+/**
+ * Clip and split a rectangle into rectangles which are not covered by the
+ * hole rectangle.
+ *
+ * @param rect
+ *     The rectangle to be split
+ *
+ * @param hole
+ *     The rectangle which represents the hole
+ *
+ * @param split_rect
+ *     Resulting split rectangle
+ *
+ * @return
+ *     Zero when no splits were done, non-zero when the rectangle was split.
+ */
+int guac_common_rect_clip_and_split(guac_common_rect* rect,
```

Comment:
I'm a broken record at this point but:

Please clarify. I understand now how this is needed and can be used, but the function documentation doesn't make that use clear, particularly the fact that it must be called multiple times.

The fact that splitting a rectangle around some arbitrary "hole" can result in multiple rectangles, only one of which will be produced here, and that further rectangles will be returned by future calls needs to be clarified.

Same deal with the fact that `rect` is modified.